paraatm/HPMM/hpmm.py

```diff
@@ -0,0 +1,101 @@
+#! /usr/bin/python3 python3
+# -*- coding: utf-8 -*-
+
+"""
+@Author: Jiawei Chen
+@Date: 2020-05-20
+This Python script is used for HPMM (Human Performance Monitoring Module)
+The script outputs predicted loss of separation as an indicator of air traffic controller's performance
+The script can also check compliance for each command
+@Last Modified by: Jiawei Chen
+@Last Modified date: 2020-05-26
+"""
+from typing import Any, Union
+
+from jpype import *
+from array import *
+import pandas as pd
+import numpy as np
+import scipy.stats as sts
+import os
+import time
+import sys
+import datetime
+from shutil import copyfile
+from shutil import rmtree
+
+import matplotlib.pyplot as plt
+from pandas import Series, DataFrame
+from pandas.io.parsers import TextFileReader
+from scipy.stats import norm
+from sklearn.preprocessing import StandardScaler
+from sklearn.model_selection import train_test_split
+import catboost as cb
+from catboost import CatBoostRegressor
+from catboost import Pool
+
+
+
+class HPMM(object):
+    def __init__(self, cfg):
+        self.data = pd.read_csv(cfg['datafile']);
+
+    def preprocessing(self):
+        # according to experiment design, remove redundant variables
+        data = self.data
+        data_domain = data.drop(
+            columns=['Ss', 'condtn', 'los_dur_over5min', 'query_timed_out', 'ready_timed_out', 'ready_latency_adj',
+                     'cum_los_dur',
+                     'stimuli', 'response_text', 'condtn_num'])
+        # fill in missing values
+        data_domain['sa_correct'].fillna(data_domain['sa_correct'].mode()[0], inplace=True)
+        data_domain.fillna(data_domain.mean(), inplace=True)
+
+        # transform categorical data
+        data_domain = data_domain.join(pd.get_dummies(data_domain.sa_correct))
+        data_domain = data_domain.join(pd.get_dummies(data_domain['query']))
+        data_domain = data_domain.drop(columns=['query', 'sa_correct'])
+        return data_domain
+
+    def modelpredict(model, input):
+        return model.predict(input)
```

Comment:
Please follow the convention of naming the first method argument `self`.  Also, I haven't tried to test the code yet, but I would be surprised if it runs: the `HPMM` class does not have a `predict` method, so it's unclear what should happen here when `modelpredict` is called on an instance of the `HPMM` class.